### PR TITLE
Increase `state_notifier` minimum version to `7.2.0`

### PR DESCRIPTION
**Pull request by ValentinVignal**
`riverpod` uses `updateShouldNotify` but it was only introduced in the version `7.2.0` of `state_notifier`

### CODE REVIEW

File: packages/flutter_riverpod/pubspec.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: flutter_riverpod
 description: >
   A simple way to access state from anywhere in your application
   while robust and testable.
-version: 2.1.1
+version: 2.1.2
```
Comment (rrousselGit) [inline packages/flutter_riverpod/pubspec.yaml:5]: Don't bump versions. You're making my life harder :)

File: website/i18n/ja/docusaurus-plugin-content-docs/current/getting_started.mdx [hunk 2]
```diff
@@ -50,7 +50,7 @@ dependencies:
   flutter:
     sdk: flutter
   flutter_hooks: ^0.18.0
-  hooks_riverpod: ^2.0.0-dev.9
+  hooks_riverpod: ^2.1.2
```
Comment (rrousselGit) [inline website/i18n/ja/docusaurus-plugin-content-docs/current/getting_started.mdx:53]: That's fine, but this make it 2.1.1 instead (after removing the version bump)

File: packages/flutter_riverpod/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 2.1.2
```
Comment (rrousselGit) [inline packages/flutter_riverpod/CHANGELOG.md:1]:
```suggestion
## [Unreleased minor]
```

File: packages/hooks_riverpod/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## 2.1.2
```
Comment (rrousselGit) [inline packages/hooks_riverpod/CHANGELOG.md:1]:
```suggestion
## [Unreleased patch]
```

File: packages/riverpod/CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,7 @@
+## 2.1.2
```
Comment (rrousselGit) [inline packages/riverpod/CHANGELOG.md:1]:
```suggestion
## [Unreleased patch]
```

**Comment by ValentinVignal** [comment]
@rrousselGit I've reverted the version bump in [⏪ Remove version bump](https://github.com/rrousselGit/riverpod/pull/1914/commits/e4a0efb772323d156e2516b47f0eafd798d21e43)

**Comment by ValentinVignal** [comment]
I'm not too sure about the 5 failing checks, are they because of my changes?

**Comment by rrousselGit** [comment]
No don't worry :)

LGTM!

**Pull request merged**